data/data/gcp/iam/main.tf

```diff
@@ -1,14 +1,58 @@
+locals {
+  # Mint Mode is default so assume Mint unless explicitly set otherwise.
+  mint_mode = var.credentials_mode != "Passthrough" && var.credentials_mode != "Manual"
+}
+
 resource "google_service_account" "worker-node-sa" {
-  account_id   = "${var.cluster_id}-w"
+  count        = local.mint_mode ? 1 : 0
```

Comment:
Would it be cleaner to put a count on the module in main.tf rather than on every resource in the module?